### DISCUSSION TITLE
How can I determine whether or not ResponseBodyTimeoutLayer timed out?

### DISCUSSION THREAD

**Discussion by mherrerarendon** [tower-rs/tower-http#536] [Q&A] [answered]
### Summary

I'm building an application using axum and I was hoping to use `ResponseBodyTimeoutLayer`.
I tried something like the following
```rust
Router::new()
    ...
    .layer(ResponseBodyTimeoutLayer::new(Duration::from_nanos(1)))
```
which does timeout the body, but my response status code is unchanged. I'm trying to figure out how to handle the body timeout (change status, etc), but I'm not having any luck. 
Any help is appreciated, thanks in advance!

### tower-http version

0.6.1

**Comment by jplatte** [comment, marked as answer]
This may not be all that helpful, but note that the status code is already sent by the time the body timeout happens. That's why it's unchanged.

**Reply by mherrerarendon** [reply]
Oh, you are right. Thank you

**Comment by mherrerarendon** [comment]
It would be impossible to modify the status if the body times out since the status gets sent first
